rasterio/_io.pyx

```diff
@@ -660,7 +698,7 @@ cdef class DatasetReaderBase(DatasetBase):
         warnings.warn(
             "read_mask() is deprecated and will be removed by Rasterio 1.0. "
             "Please use read_masks() instead.",
-            FutureWarning,
+            DeprecationWarning,
```

Comment:
@sgillies `FutureWarning` is not filtered by default, whereas `DeprecationWarning` is (meaning it isn't visible to most users).  I'm assuming you want this visible by default?

If so:
```
with warnings.catch_warnings():
    warnings.simplefilter('always')
    warnings.warn(...)
```